presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -257,6 +270,13 @@ public NodeSelector createNodeSelector(Session session, ConnectorId connectorId,
                     .map(InternalNode::getNodeIdentifier)
                     .collect(toImmutableSet());
 
+            int weight = (int) ceil(1.0 * minVirtualNodeCount / activeNodes.size());
+            for (InternalNode node : activeNodes) {
+                for (int i = 0; i < weight; i++) {
+                    activeNodesByConsistentHashing.put(murmur3_32().hashString(format("%s%d", node.getNodeIdentifier(), i), UTF_8).asInt(), node);
+                }
+            }
```

Comment:
We should move those logic inside `ConsistentHashingNodeProvider` given they're specific to `ConsistentHashingNodeProvider` and NodeScheduler shouldn't be aware of it

Reply:
In that case the map will be created for every query. Here it's created only once when node map is updated.

Reply:
That's a good point, then should we actually only have one  `NodeProvider ` in the system instead of creating one for each query

Reply:
I thought about that. Basically we only need to create a `NodeProvider` when `NodeMap` is changed. Currently this is cumbersome to do because when we use soft affinity with modular hashing, we are using all nodes, rather than just active nodes, to reduce reshuffling due to temporary worker unavailability.  Once re remove modular hashing, and always use active nodes as candidate nodes, the `NodeProvider` logic can be moved into `NodeMap` entirely.